tests/transformers/test_transformers_model_export.py

```diff
@@ -3965,3 +3965,25 @@ def test_text_generation_task_completions_serve(text_generation_pipeline):
     assert output_dict["text"] is not None
     assert output_dict["finish_reason"] == "stop"
     assert output_dict["usage"]["prompt_tokens"] < 20
+
+
+def test_model_config_is_not_mutated_after_prediction(text2text_generation_pipeline):
+    model_config = {
+        "top_k": 2,
+        "num_beams": 5,
+        "max_length": 30,
+    }
+    # Params will be used to override the values of model_config but should not mutate it
+    params = {
+        "top_k": 3,
+        "max_length": 50,
+    }
+
+    pyfunc_model = _TransformersWrapper(text2text_generation_pipeline, model_config=model_config)
+    assert pyfunc_model.model_config["top_k"] == 2
+
+    pyfunc_model.predict("How to learn Python in 3 weeks?", params=params)
+
+    assert pyfunc_model.model_config["top_k"] == 2
+    assert pyfunc_model.model_config["num_beams"] == 5
+    assert pyfunc_model.model_config["max_length"] == 30
```

Comment:
We could get some assured validation that an overridden parameter behaves how we expect by forcing a brief generation output (the underlying model that is used for this pipeline fixture is a generator from keywords) by generating more keywords than are permitted to construct a sentence using all of them by restricting the `max_new_tokens` parameter.

```suggestion
    model_config = {
        "top_k": 2,
        "num_beams": 5,
        "max_length": 30,
        "max_new_tokens": 500,
    }
    # Params will be used to override the values of model_config but should not mutate it
    params = {
        "top_k": 30,
        "max_length": 500,
        "max_new_tokens": 5,
    }

    pyfunc_model = _TransformersWrapper(text2text_generation_pipeline, model_config=model_config)
    assert pyfunc_model.model_config["top_k"] == 2

    prediction_output = pyfunc_model.predict(
        "rocket moon ship astronaut space gravity", params=params
    )

    assert pyfunc_model.model_config["top_k"] == 2
    assert pyfunc_model.model_config["num_beams"] == 5
    assert pyfunc_model.model_config["max_length"] == 30
    assert pyfunc_model.model_config["max_new_tokens"] == 500
    assert len(prediction_output[0].split(" ")) <= 5
```

Reply:
wow neat way for the validation, thanks!